### PR TITLE
Issue #10752: Added support for violation N lines above/below in Inline Config Parser

### LINKED ISSUE #10752: Add support for violation N lines above/below in Inline Config Parser
inspired from https://github.com/checkstyle/checkstyle/issues/10692#issuecomment-907672014

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #10752: Added support for violation N lines above/below in Inline Config Parser
Issue #10692: Migrate to Inline Config Parser in Javadoc Check Tests

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java [hunk 1]
```diff
@@ -257,52 +265,72 @@ else if (value.startsWith("(default)")) {
...
-                        .nCopies(Integer.parseInt(multipleViolationsBelowMatcher.group(1)),
-                                lineNo + 2)
-                        .forEach(actualLineNumber -> {
-                            inputConfigBuilder.addViolation(actualLineNumber, null);
-                        });
-            }
-            else if (useFilteredViolations) {
-                setFilteredViolation(inputConfigBuilder, lineNo + 1, lines.get(lineNo));
-            }
+            setViolations(inputConfigBuilder, lines, useFilteredViolations, lineNo);
```
Comment (Vyom-Yadav) [inline src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java:268]: Extracted to reduce complexity.

**Comment by romani** [comment]
Please make two commit in this PR.
First commit, extraction to method.
Second commit, functional change.

**Comment by Vyom-Yadav** [comment]
@romani done.

**Comment by Vyom-Yadav** [comment]
Also something like `// multiple violation X lines above` can be added, but I think it'll be overkill as we can perform unit testing without it.

**Comment by romani** [comment]
Please make two commit in this PR.
First commit, extraction to method with message "minor: extraction of method of future extension in Issue #xxxxx".
Second commit, functional change with message `Issue #10752: Added support for violation N lines above/below in Inline Config Parser` where you make code changes and migrate Inputs to it.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheckTest.java [hunk 2]
```diff
@@ -19,13 +19,12 @@
 
 package com.puppycrawl.tools.checkstyle.checks.javadoc;
 
+import static com.google.common.truth.Truth.assertWithMessage;
 import static com.puppycrawl.tools.checkstyle.checks.javadoc.NonEmptyAtclauseDescriptionCheck.MSG_KEY;
-import static org.junit.jupiter.api.Assertions.assertArrayEquals;
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheckTest.java:24]: please move migration to Truth to separate commit.
Do not mix "Inline Config" and "Migration to Truth" in same commit.

File: config/suppressions.xml [hunk 3]
```diff
@@ -418,8 +418,6 @@
             files="[\\/]src[\\/]test[\\/]java[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]MissingJavadocPackageCheckTest.java"/>
   <suppress id="ImportControlTest"
             files="[\\/]src[\\/]test[\\/]java[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]MissingJavadocTypeCheckTest.java"/>
-  <suppress id="ImportControlTest"
-            files="[\\/]src[\\/]test[\\/]java[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]checks[\\/]javadoc[\\/]NonEmptyAtclauseDescriptionCheckTest.java"/>
```
Comment (romani) [inline config/suppressions.xml:422]: move "Infra: migrate to Truth in tests " to separate commit of this PR or separate PR.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheckTest.java (hunk 2 above)
Comment (Vyom-Yadav) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheckTest.java:24]: done

File: config/suppressions.xml (hunk 3 above)
Comment (Vyom-Yadav) [reply config/suppressions.xml:422]: done

**Comment by Vyom-Yadav** [comment]
@romani As you directed, I have made 3 commits in this PR-

1. extraction of method
2. Add support for N lines above/below and migrate to inline config parser
3. migrate to truth in tests

**Review by romani — APPROVED** [review]
Ok to merge

**Review by pbludov — APPROVED** [review]

**Review by rnveach — APPROVED** [review]
I am approving to merge, but I don't find the new input file easy to read at all. I don't have an alternative to provide at this time.
